packages/contracts/slither.config.json

```diff
@@ -1,7 +1,7 @@
 {
   "exclude_informational": true,
   "exclude_low": true,
-  "detectors_to_exclude": "naming-conventions,different-pragma-directives-are-used,external-function,assembly,incorrect-equality,incorrect-versions-of-solidity,unused-return",
+  "detectors_to_exclude": "naming-conventions,different-pragma-directives-are-used,external-function,assembly,incorrect-equality,incorrect-versions-of-solidity,unused-return,state-variables-that-could-be-declared-constant",
```

Comment:
> state-variables-that-could-be-declared-constant

What is this doing? Why shouldn't state variables normally be declared as constants?

Reply:
like i did make the most of the static variables constant but like there was these two on making them constant there was errors about "value should be known at compile time" so i had to use that to avoid it we can fix it if u want 
 uint256 deployerPrivateKey = vm.envUint("PRIVATE_KEY");
 address admin = vm.envAddress("PUBLIC_KEY");

Reply:
all though on adding it was still giving the error as before but it was in blue so can be taken as suggestion

Reply:
Perhaps we can initialize these values with `0` address and throw an error if they are the `0` address. That way during runtime we should never have an issue, and Slither might stop complaining?

Reply:
> Perhaps we can initialize these values with `0` address and throw an error if they are the `0` address. That way during runtime we should never have an issue, and Slither might stop complaining?

oh sure we can try

Reply:
> Perhaps we can initialize these values with `0` address and throw an error if they are the `0` address. That way during runtime we should never have an issue, and Slither might stop complaining?

but like if we initialize the variable which is constant to 0 how we gonna change it 🤔

Reply:
@pavlovcik i used immutable and it seemss to worked should we go with it?

Reply:
slither not complaining too and also removed the "state-variables-that-could-be-declared-constant"

Reply:
I'm not a Solidity developer so I don't know.

Reply:
> I'm not a Solidity developer so I don't know.

oh using the immutable then constant is like immutable's can be changed only using Constructors making it kinda constant but  slither seems to have to no problem with it

Reply:
> @pavlovcik i used immutable and it seemss to worked should we go with it?

@rndquu

Reply:
@AnakinSkywalkeer I can't find the `state-variables-that-could-be-declared-constant` detector docs [here](https://github.com/crytic/slither/wiki/Detector-Documentation). There is [one](https://github.com/crytic/slither/wiki/Detector-Documentation#state-variables-that-could-be-declared-constant) called `constable-states` which seems to do the same task. Could you share a URL for the `state-variables-that-could-be-declared-constant` detector docs?

Reply:
> @AnakinSkywalkeer I can't find the `state-variables-that-could-be-declared-constant` detector docs [here](https://github.com/crytic/slither/wiki/Detector-Documentation). There is [one](https://github.com/crytic/slither/wiki/Detector-Documentation#state-variables-that-could-be-declared-constant) called `constable-states` which seems to do the same task. Could you share a URL for the `state-variables-that-could-be-declared-constant` detector docs?

for sure , [here](https://github.com/crytic/slither/wiki/Detector-Documentation#state-variables-that-could-be-declared-constant)

Reply:
The detector you mentioned in the URL is called `constable-states` while in the code the `state-variables-that-could-be-declared-constant` detector is used which is basically an invalid detector name. Perhaps we can remove the `state-variables-that-could-be-declared-constant` detector from the exclude list?

Reply:
> The detector you mentioned in the URL is called `constable-states` while in the code the `state-variables-that-could-be-declared-constant` detector is used which is basically an invalid detector name. Perhaps we can remove the `state-variables-that-could-be-declared-constant` detector from the exclude list?

yeah sure